### PR TITLE
unbound: use $USER in unbound.conf

### LINKED ISSUE #40110: Unbound crashes immediately on post install startup
I tried installing Unbound today and followed the post install launchd startup instructions. The application crashes on my Yosemite system immediately on startup.  Crash log is below.

https://github.com/Homebrew/homebrew/blob/master/Library/Formula/unbound.rb

````
~$ cat ~/Desktop/unbound_2015-05-26-131552_MYHOSTNAME.crash
Process:               unbound [43266]
Path:                  /usr/local/Cellar/unbound/1.5.3/sbin/unbound
Identifier:            unbound
Version:               0
Code Type:             X86-64 (Native)
Parent Process:        launchd [1]
Responsible:           unbound [43266]
User ID:               0

Date/Time:             2015-05-26 13:15:52.418 -0700
OS Version:            Mac OS X 10.10.3 (14D136)
Report Version:        11
Anonymous UUID:        60DC6375-8469-4E2E-EA8A-3ACE84340883

Sleep/Wake UUID:       08AA4AD0-E1CB-4ED3-8718-F3B55D54E51F

Time Awake Since Boot: 2900000 seconds
Time Since Wake:       1700000 seconds

Crashed Thread:        0  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_BAD_ACCESS (SIGSEGV)
Exception Codes:       KERN_INVALID_ADDRESS at 0x0000000000000010

VM Regions Near 0x10:
-->
    __TEXT                 000000010a701000-000000010a785000 [  528K] r-x/rwx SM=COW  /usr/local/Cellar/unbound/1.5.3/sbin/unbound

Thread 0 Crashed:: Dispatch queue: com.apple.main-thread
0   unbound                       	0x000000010a70e267 main + 4585
1   libdyld.dylib                 	0x00007fff817ef5c9 start + 1

Thread 0 crashed with X86 Thread State (64-bit):
  rax: 0x30002ef513f05081  rbx: 0x00007f8ac14210f0  rcx: 0x0000000000000000  rdx: 0x000000010a7893d8
  rdi: 0x0000010000000203  rsi: 0x0000020000000200  rbp: 0x00007fff554fee30  rsp: 0x00007fff554fed00
   r8: 0x00007fff554fc320   r9: 0x00007fff710db300  r10: 0x0000000000000000  r11: 0x0000000000000246
  r12: 0x00007f8ac1404830  r13: 0x000000000000102a  r14: 0x0000000000000000  r15: 0x00007f8ac1404800
  rip: 0x000000010a70e267  rfl: 0x0000000000010206  cr2: 0x0000000000000010

Logical CPU:     2
Error Code:      0x00000004
Trap Number:     14


Binary Images:
       0x10a701000 -        0x10a784fff +unbound (0) <606BFF8D-6D40-34BE-8D8E-21E43A4FC1FC> /usr/local/opt/unbound/sbin/unbound
       0x10a798000 -        0x10a7d5fff +libssl.1.0.0.dylib (0) <744C7DDF-7960-3BEF-B4FB-510B1E293D33> /usr/local/opt/openssl/lib/libssl.1.0.0.dylib
       0x10a7f2000 -        0x10a81aff7 +libevent-2.0.5.dylib (0) <C1B544F2-35D7-371B-A607-3B8D1092FBFB> /usr/local/lib/libevent-2.0.5.dylib
       0x10a828000 -        0x10a99849f +libcrypto.1.0.0.dylib (0) <4EA19050-1F9F-3797-A385-B52C85A5AEA5> /usr/local/opt/openssl/lib/libcrypto.1.0.0.dylib
       0x10ae3f000 -        0x10ae4bff7 +libgost.dylib (0) <E32EA54A-87EA-3BE1-B10E-E45E265EDE3F> /usr/local/Cellar/openssl/1.0.2a-1/lib/engines/libgost.dylib
    0x7fff6eb1e000 -     0x7fff6eb54837  dyld (353.2.1) <65DCCB06-339C-3E25-9702-600A28291D0E> /usr/lib/dyld
    0x7fff817ec000 -     0x7fff817efff7  libdyld.dylib (353.2.1) <9EACCA38-291D-38CC-811F-7E9D1451E2D3> /usr/lib/system/libdyld.dylib
    0x7fff81919000 -     0x7fff81921fff  libsystem_platform.dylib (63) <64E34079-D712-3D66-9CE2-418624A5C040> /usr/lib/system/libsystem_platform.dylib
    0x7fff819de000 -     0x7fff819e0fff  libquarantine.dylib (76.20.1) <7AF90041-2768-378A-925A-D83161863642> /usr/lib/system/libquarantine.dylib
    0x7fff8200e000 -     0x7fff8200fff3  libSystem.B.dylib (1213) <CCEC13A5-D0D9-31C5-B0B0-1C564B4A20A6> /usr/lib/libSystem.B.dylib
    0x7fff82260000 -     0x7fff8227cff7  libsystem_malloc.dylib (53.1.1) <19BCC257-5717-3502-A71F-95D65AFA861B> /usr/lib/system/libsystem_malloc.dylib
    0x7fff82a0c000 -     0x7fff82a34fff  libxpc.dylib (559.20.9) <D35D0DB2-D7BD-3BE4-8378-062BFE545E1D> /usr/lib/system/libxpc.dylib
    0x7fff82c39000 -     0x7fff82c3bfff  libsystem_configuration.dylib (699.1.5) <20F3B077-179D-3CB0-A3C1-C8602D53B4DB> /usr/lib/system/libsystem_configuration.dylib
    0x7fff82f85000 -     0x7fff82fb5fff  libsystem_m.dylib (3086.1) <1E12AB45-6D96-36D0-A226-F24D9FB0D9D6> /usr/lib/system/libsystem_m.dylib
    0x7fff82fc2000 -     0x7fff82fd3fff  libsystem_coretls.dylib (35.20.2) <6084A531-2523-39F8-B030-811FA1A32FB5> /usr/lib/system/libsystem_coretls.dylib
    0x7fff82fd5000 -     0x7fff82fdafff  libsystem_stats.dylib (163.20.16) <FBC3F80F-A0FB-3BD6-9A7E-800DE45F092E> /usr/lib/system/libsystem_stats.dylib
    0x7fff82fdb000 -     0x7fff82fe4fff  libsystem_pthread.dylib (105.10.1) <3103AA7F-3BAE-3673-9649-47FFD7E15C97> /usr/lib/system/libsystem_pthread.dylib
    0x7fff832d6000 -     0x7fff832d7fff  libDiagnosticMessagesClient.dylib (100) <2EE8E436-5CDC-34C5-9959-5BA218D507FB> /usr/lib/libDiagnosticMessagesClient.dylib
    0x7fff834bf000 -     0x7fff834c4ff7  libmacho.dylib (862) <126CA2ED-DE91-308F-8881-B9DAEC3C63B6> /usr/lib/system/libmacho.dylib
    0x7fff8355f000 -     0x7fff8356afff  libcommonCrypto.dylib (60061) <D381EBC6-69D8-31D3-8084-5A80A32CB748> /usr/lib/system/libcommonCrypto.dylib
    0x7fff836ff000 -     0x7fff8372afff  libc++abi.dylib (125) <88A22A0F-87C6-3002-BFBA-AC0F2808B8B9> /usr/lib/libc++abi.dylib
    0x7fff83795000 -     0x7fff8379dfff  libsystem_dnssd.dylib (561.1.1) <62B70ECA-E40D-3C63-896E-7F00EC386DDB> /usr/lib/system/libsystem_dnssd.dylib
    0x7fff83ae7000 -     0x7fff83b5dfe7  libcorecrypto.dylib (233.1.2) <E1789801-3985-3949-B736-6B3378873301> /usr/lib/system/libcorecrypto.dylib
    0x7fff85368000 -     0x7fff85368ff7  libunc.dylib (29) <5676F7EA-C1DF-329F-B006-D2C3022B7D70> /usr/lib/system/libunc.dylib
    0x7fff86543000 -     0x7fff86554ff7  libz.1.dylib (55) <88C7C7DE-04B8-316F-8B74-ACD9F3DE1AA1> /usr/lib/libz.1.dylib
    0x7fff87082000 -     0x7fff87084ff7  libsystem_coreservices.dylib (9) <41B7C578-5A53-31C8-A96F-C73E030B0938> /usr/lib/system/libsystem_coreservices.dylib
    0x7fff870ab000 -     0x7fff870b0ff7  libunwind.dylib (35.3) <BE7E51A0-B6EA-3A54-9CCA-9D88F683A6D6> /usr/lib/system/libunwind.dylib
    0x7fff87629000 -     0x7fff8762fff7  libsystem_networkextension.dylib (167.1.10) <29AB225B-D7FB-30ED-9600-65D44B9A9442> /usr/lib/system/libsystem_networkextension.dylib
    0x7fff87736000 -     0x7fff8773effb  libcopyfile.dylib (118.1.2) <0C68D3A6-ACDD-3EF3-991A-CC82C32AB836> /usr/lib/system/libcopyfile.dylib
    0x7fff87b65000 -     0x7fff87b65ff7  libkeymgr.dylib (28) <77845842-DE70-3CC5-BD01-C3D14227CED5> /usr/lib/system/libkeymgr.dylib
    0x7fff87bd1000 -     0x7fff87bd2ff7  libsystem_blocks.dylib (65) <9615D10A-FCA7-3BE4-AA1A-1B195DACE1A1> /usr/lib/system/libsystem_blocks.dylib
    0x7fff87bd3000 -     0x7fff87bd9fff  libsystem_trace.dylib (72.20.1) <840F5301-B55A-3078-90B9-FEFFD6CD741A> /usr/lib/system/libsystem_trace.dylib
    0x7fff88656000 -     0x7fff8865afff  libcache.dylib (69) <45E9A2E7-99C4-36B2-BEE3-0C4E11614AD1> /usr/lib/system/libcache.dylib
    0x7fff8865b000 -     0x7fff88693fff  libsystem_network.dylib (412.20.3) <589A5F67-BE2A-3245-A181-0ECC9B53EB00> /usr/lib/system/libsystem_network.dylib
    0x7fff88694000 -     0x7fff886b1fff  libsystem_kernel.dylib (2782.20.48) <EAFD7BD0-0C30-3E7D-9528-F9916BA0167C> /usr/lib/system/libsystem_kernel.dylib
    0x7fff88728000 -     0x7fff88729ffb  libremovefile.dylib (35) <3485B5F4-6CE8-3C62-8DFD-8736ED6E8531> /usr/lib/system/libremovefile.dylib
    0x7fff8949b000 -     0x7fff894a4ff7  libsystem_notify.dylib (133.1.1) <61147800-F320-3DAA-850C-BADF33855F29> /usr/lib/system/libsystem_notify.dylib
    0x7fff894a5000 -     0x7fff894cdfff  libsystem_info.dylib (459.20.1) <AEB3FE62-4763-3050-8352-D6F9AF961AE6> /usr/lib/system/libsystem_info.dylib
    0x7fff89a3a000 -     0x7fff89c3446f  libobjc.A.dylib (647) <759E155D-BC42-3D4E-869B-6F57D477177C> /usr/lib/libobjc.A.dylib
    0x7fff8bdf3000 -     0x7fff8be1dff7  libdispatch.dylib (442.1.4) <502CF32B-669B-3709-8862-08188225E4F0> /usr/lib/system/libdispatch.dylib
    0x7fff8c84a000 -     0x7fff8c851ff7  libcompiler_rt.dylib (35) <BF8FC133-EE10-3DA6-9B90-92039E28678F> /usr/lib/system/libcompiler_rt.dylib
    0x7fff8cddd000 -     0x7fff8ce23ff7  libauto.dylib (186) <A260789B-D4D8-316A-9490-254767B8A5F1> /usr/lib/libauto.dylib
    0x7fff8ce24000 -     0x7fff8ce25fff  libsystem_secinit.dylib (18) <581DAD0F-6B63-3A48-B63B-917AF799ABAA> /usr/lib/system/libsystem_secinit.dylib
    0x7fff8cf64000 -     0x7fff8cf7aff7  libsystem_asl.dylib (267) <F153AC5B-0542-356E-88C8-20A62CA704E2> /usr/lib/system/libsystem_asl.dylib
    0x7fff8e41a000 -     0x7fff8e4a6ff7  libsystem_c.dylib (1044.10.1) <86FBED7A-F2C8-3591-AD6F-486DD57E6B6A> /usr/lib/system/libsystem_c.dylib
    0x7fff8ef69000 -     0x7fff8efbdfff  libc++.1.dylib (120) <1B9530FD-989B-3174-BB1C-BDC159501710> /usr/lib/libc++.1.dylib
    0x7fff8ffe4000 -     0x7fff8ffe6fff  libsystem_sandbox.dylib (358.20.5) <4CF77128-6BE0-3958-B646-707FA9CE61B2> /usr/lib/system/libsystem_sandbox.dylib
    0x7fff900f5000 -     0x7fff900f5ff7  liblaunch.dylib (559.20.9) <FA89A113-696E-3271-8FE1-A0D7324E8481> /usr/lib/system/liblaunch.dylib

External Modification Summary:
  Calls made by other processes targeting this process:
    task_for_pid: 0
    thread_create: 0
    thread_set_state: 0
  Calls made by this process:
    task_for_pid: 0
    thread_create: 0
    thread_set_state: 0
  Calls made by all processes on this machine:
    task_for_pid: 42747431
    thread_create: 0
    thread_set_state: 0

VM Region Summary:
ReadOnly portion of Libraries: Total=79.6M resident=79.3M(100%) swapped_out_or_unallocated=240K(0%)
Writable regions: Total=23.9M written=1232K(5%) resident=2480K(10%) swapped_out=0K(0%) unallocated=21.5M(90%)

REGION TYPE                      VIRTUAL
===========                      =======
Kernel Alloc Once                     4K
MALLOC                             11.4M
MALLOC (admin)                       16K
SBRK (reserved)                    4096K        reserved VM address space (unallocated)
STACK GUARD                        56.0M
Stack                              8192K
VM_ALLOCATE                          12K
__DATA                              896K
__LINKEDIT                         71.3M
__TEXT                             8484K
shared memory                         4K
===========                      =======
TOTAL                             159.9M
TOTAL, minus reserved VM space    155.9M

````

### PR DESCRIPTION
**Pull request by dunn**
Fixes #40110.

### CODE REVIEW

**Comment by xu-cheng** [comment]
Should we do this in `post_install`? Otherwise, it will be used as bot's name for bottles?

**Comment by dunn** [comment]
Oh, you're right!

File: Library/Formula/unbound.rb [hunk 1]
```diff
@@ -52,4 +57,9 @@ def plist; <<-EOS.undent
     </plist>
     EOS
   end
+
+  def caveats; <<-EOS.undent
+    A sample configuration file has been installed in #{etc}/unbound.
+  EOS
+  end
```
Comment (mikemcquaid) [inline Library/Formula/unbound.rb]: Let's just kill these.

File: Library/Formula/unbound.rb [hunk 2]
```diff
@@ -19,9 +19,14 @@ def install
                           "--sysconfdir=#{etc}",
                           "--with-libevent=#{Formula["libevent"].opt_prefix}",
                           "--with-ssl=#{Formula["openssl"].opt_prefix}"
+
```
Comment (mikemcquaid) [inline Library/Formula/unbound.rb]: Can kill the blank line here unless you much prefer it.

**Comment by mikemcquaid** [comment]
:+1:

**Comment by dunn** [comment]
If we uncomment the `username` line in `unbound.conf`, we can also add a basic test without it complaining about there being no "unbound" user.  But is that too invasive?

**Comment by xu-cheng** [comment]
Can you instead copy the conf file to testpath, do the alteration and feed it back to the program (maybe by argument or environment variable)?

**Comment by dunn** [comment]
That gives this error:
```
[1439134496] unbound-checkconf[31647:0] fatal error: config file /private/tmp/unbound20150809-31637-ggf06i/unbound.conf is not inside chroot /usr/local/etc/unbound
```
So I found a better test instead.

**Pull request closed without merging**